shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -378,6 +444,36 @@ - (void)setupNotificationCenterObservers {
                object:nil];
 }
 
+- (void)setupKeyboardAnimationCurve {
+  _keyboardAnimationStops.reset([[NSMutableArray alloc] init]);
+
+  // Keyboard animation spring curve for iOS 16 and below
+  _keyboardSpringCurve.reset([[KeyboardSpringCurve alloc] initWithStiffness:1000
+                                                               dampingRatio:1
+                                                                       mass:3
+                                                            initialVelocity:0
+                                                           settlingDuration:0.5]);
+  double settlingDuration = [[self keyboardSpringCurve] settlingDuration];
+  double maxFramesPerSecond = [DisplayLinkManager displayRefreshRate];
+  double frameDuration = 1 / maxFramesPerSecond;
+
+  // Calculating a full seconds worth of animation stops for increased stop fidelity
+  for (int time = 1; time < maxFramesPerSecond; time++) {
+    double frameTime = time / maxFramesPerSecond;
+    NSNumber* keyboardAnimationStop =
+        [NSNumber numberWithDouble:[[self keyboardSpringCurve] curveFunc:frameTime]];
+
+    // Only store animation stops up to the settlingDuration plus one (since time starts at 1)
+    if (frameTime > settlingDuration + frameDuration) {
+      // Ensure the animation finishes with 100%
+      [[self keyboardAnimationStops] addObject:[NSNumber numberWithDouble:1]];
+      break;
+    }
```

Comment:
We can move this in front of the line below and we can avoid invaild compute once for last loop.^_^
```
NSNumber* keyboardAnimationStop =
        [NSNumber numberWithDouble:[[self keyboardSpringCurve] curveFunc:frameTime]];
```

Reply:
Good catch :)

Reply:
1. Nits: Add "." for all comments.(*^▽^*)

2. Why does loop start from `time = 1` instead of `time = 0`🤔?